sys/boot/geli/geliboot.c

```diff
@@ -217,20 +268,22 @@ geli_attach(struct dsk *dskp, const char *passphrase)
 		error = g_eli_mkey_decrypt(&geli_e->md, key, mkey, &keynum);
 		if (error == -1) {
 			bzero(&mkey, sizeof(mkey));
-                        bzero(&key, sizeof(key));
-			printf("Bad GELI key: %d\n", error);
+			bzero(&key, sizeof(key));
+			printf("Bad GELI key: bad password?\n");
 			return (error);
 		} else if (error != 0) {
 			bzero(&mkey, sizeof(mkey));
-                        bzero(&key, sizeof(key));
+			bzero(&key, sizeof(key));
                         printf("Failed to decrypt GELI master key: %d\n", error);
 			return (error);
 		} else {
-                        /* Store the keys */
+                        /* Add key to keychain */
```

Comment:
'Stores the Keys' was moved by your previous commit.
It is actually about loading the mkey and ivkey into the geli_e struct, so i reverted that and added a separate comment for stashing the userkey to avoid having to regenerate it later